src/test/scala/karazin/scala/users/group/week2/homework/HomeworkSpecification.scala

```diff
@@ -49,29 +43,31 @@ object HomeworkSpecification extends Properties("Homework"):
     (left >= right) == ( left > right || left == right)
   }
 
-  property("negation") = forAll { (rational: Rational) =>
-    ???
+  property("addition") = forAll { (left: Rational, right: Rational) =>
+    (Rational(left.numer * right.denom + right.numer * left.denom, left.denom * right.denom) == (left + right))
   }
 
-  property("addition") = forAll { (left: Rational, right: Rational) =>
-    ???
+  property("negation") = forAll { (rational: Rational) =>
+    (Rational((-1) * rational.numer, rational.denom) == (-rational))
   }
 
   property("subtraction") = forAll { (left: Rational, right: Rational) =>
-    ???
+    (Rational(left.numer * right.denom - right.numer * left.denom, left.denom * right.denom) == (left - right))
   }
 
   property("multiplication") = forAll { (left: Rational, right: Rational) =>
-    ???
+    (Rational(left.numer * right.numer, left.denom * right.denom) == (left * right))
   }
 
   property("division") = forAll { (left: Rational, numer: Int, denom: Int) =>
     val right = Rational(if numer == 0 then 1 else numer, abs(denom) + 1)
-    ???
+    Rational(signum(right.numer) * left.numer * right.denom, abs(left.denom * right.numer)) == (left / right)
   }
 
   property("division by zero") = forAll { (left: Rational, int: Int) =>
-    ???
+    throws(classOf[IllegalArgumentException]) {
+      left / Rational(int, 0)
```

Comment:
It's not a division by zero rational. It's an attempt of creating Rational with zero denom. Please fix it.